packages/ember-metal/lib/weak_map.js

```diff
@@ -57,12 +57,12 @@ WeakMap.prototype.get = function(obj) {
   let meta = peekMeta(obj);
   if (meta) {
     let map = meta.readableWeak();
-    if (map) {
-      if (map[this._id] === UNDEFINED) {
+    if (map !== undefined) {
```

Comment:
This seems unrelated to what this PR title suggests is the goal ("reuse meta arrayContentDidChange"), can you remove that commit and submit separately?